src/app/components/Item/ItemFilters.jsx

```diff
@@ -1,27 +1,116 @@
     format: query.format || [],
     status: query.status || [],
   };
-
+  const resultsRef = useRef(null);
+  const [openFilter, setOpenFilter] = useState('none');
+  // The "year" filter is not used for the `ItemFilter` dropdown component
+  // and must be handled separately in the `SearchBar` component.
+  const [selectedYear, setSelectedYear] = useState(query.date || '');
   const [selectedFilters, setSelectedFilters] = useState(initialFilters);
+  const [selectedFilterDisplayStr, setSelectedFilterDisplayStr] = useState('');
+  // For every item aggregation, go through every filter in its `values` array
+  // and map all the labels to their ids. This is done because the API expects
+  // the ids of the filters to be sent over, not the labels.
+  const mappedItemsLabelToIds = itemsAggregations.reduce((accc, aggregation) => {
+    const filter = aggregation.field;
+    const mappedValues = aggregation.values.reduce((acc, value) => ({
+      ...acc,
+      [value.label]: value.value
+    }), {})
+    return {
+      ...accc,
+      [filter]: mappedValues,
+    };
+  }, {});
+
+  // When new items are fetched, update the selected string dispaly.
+  useEffect(() => {
+    setSelectedFilterDisplayStr(parsedFilterSelections());
+  }, [numOfFilteredItems, parsedFilterSelections])
+
+  /**
+   * When new filters are selected or unselected, fetch new items.
+   */
+  const getNewBib = (clear = false, clearYear = false) => {
+    const baseUrl = appConfig.baseUrl;
+    let queryParams = [];
+    if (!clear) {
+      // If we are making a request, get all the selected filters and map
+      // the labels to their ids. That will be sent over to the API.
+      Object.keys(selectedFilters).forEach(filter => {
+        const selectedFilterArray = selectedFilters[filter];
+        const mappedFilter = mappedItemsLabelToIds[filter];
+        if (selectedFilterArray.length > 0) {
+          if (Array.isArray(selectedFilterArray)) {
+            queryParams.push(
+              ...selectedFilterArray.map(selectedFilter => {
+                const mappedFilterId = mappedFilter[selectedFilter];
+                return `${filter}=${mappedFilterId}`;
+              })
+            );
+          } else {
+            const mappedFilterId = mappedFilter[selectedFilterArray];
+            queryParams.push(`${filter}=${mappedFilterId}`);
+          }
+        }
+      });
+      // The "year" filter is stored separately from the other filters.
+      if (selectedYear && !clearYear) {
+        queryParams.push(`date=${selectedYear}`);
+      }
+    }
+
+    const bibApi = `${window.location.pathname.replace(
+      baseUrl,
+      `${baseUrl}/api`,
+    )}${queryParams.length ? `?${queryParams.join('&')}` : ''}`;
+    // Make the API call and let redux know.
+    ajaxCall(
+      bibApi,
+      (resp) => {
+        const { bib } = resp.data;
+        const done = !bib || !bib.items || bib.items.length < itemBatchSize;
+        dispatch(
```

Comment:
why is this dispatch function being passed as a prop?

Reply:
It seemed easier than connecting the `ItemFilters` component to redux or bringing in that specific context. If we also want to move away from redux, this could be replaced more easily in the future.

Reply:
ok!